hms-app/src/components/HackathonSelectDropdown.tsx

```diff
@@ -27,38 +31,63 @@ export default function HackathonSelectDropdown(setHackathonID: Props) {
     )
   }
 
-  const hackathonMap = hackathonList.map((hackathon, index) => ({
-    value: hackathon.id,
-    label:
-      hackathon.title +
-      ' ' +
-      new Date(hackathon.startDate).toLocaleDateString() +
-      '-' +
-      new Date(hackathon.endDate).toLocaleDateString(),
-  }))
+  const futureAndPresentHackathons = hackathonList.filter((hackathon) => {
+    return new Date(hackathon.endDate) > today
+  })
 
-  const selectChange = (value: string) => {
-    setSelectedHackweek(value)
+  const pastHackathons = hackathonList.filter((hackathon) => {
+    return new Date(hackathon.endDate) < today
+  })
+
+  function getHackatonsForContext(): HackathonPreview[] {
+    switch (context) {
+      case 'archive':
+        return pastHackathons
+      case 'idea-portal':
+        return futureAndPresentHackathons
+    }
+    return hackathonList
+  }
+
+  function mapHackathonToSelectItem(hackathon: HackathonPreview): SelectItem {
+    return {
+      value: hackathon.id,
+      label:
+        hackathon.title +
+        ' ' +
+        new Date(hackathon.startDate).toLocaleString(undefined, {
+          year: 'numeric',
+          month: '2-digit',
+          day: '2-digit',
+        }) +
+        '-' +
+        new Date(hackathon.endDate).toLocaleString(undefined, {
```

Comment:
```suggestion
        hackathon.startDate.toLocaleString(undefined, {
          year: 'numeric',
          month: '2-digit',
          day: '2-digit',
        }) +
        '-' +
        ackathon.endDate.toLocaleString(undefined, {
```

Reply:
no

Reply:
endDate and startDate are already `Date` objects so you shouldn't need to convert them into another Date object

Reply:
the problem might be that when you do the fetch for the data you don't convert them to Date objects immediately.  The type defines them as being Date objects so you should always assume they are, you just have to ensure that when you create an instance of that type you set the values properly

Reply:
the problem is not the type as i understand
the problem is when we want to display them with DD.MM.YYYY this is the way we have to do it 
otherwise we would have a different format

Reply:
doing this shouldn't have any effect on the format.  That is because when you call `toLocalDateString` you force the format to be a certain way